lib/matplotlib/backend_tools.py

```diff
@@ -626,28 +625,44 @@ def scroll_zoom(self, event):
         # https://gist.github.com/tacaswell/3144287
         if event.inaxes is None:
             return
-        ax = event.inaxes
-        cur_xlim = ax.get_xlim()
-        cur_ylim = ax.get_ylim()
-        # set the range
-        cur_xrange = (cur_xlim[1] - cur_xlim[0])*.5
-        cur_yrange = (cur_ylim[1] - cur_ylim[0])*.5
-        xdata = event.xdata  # get event x location
-        ydata = event.ydata  # get event y location
+
+        scl = self.base_scale
+
         if event.button == 'up':
             # deal with zoom in
-            scale_factor = 1 / self.base_scale
+            direction = 'in'
         elif event.button == 'down':
             # deal with zoom out
-            scale_factor = self.base_scale
+            direction = 'out'
         else:
             # deal with something that should never happen
-            scale_factor = 1
-        # set new limits
-        ax.set_xlim([xdata - cur_xrange*scale_factor,
-                     xdata + cur_xrange*scale_factor])
-        ax.set_ylim([ydata - cur_yrange*scale_factor,
-                     ydata + cur_yrange*scale_factor])
+            scl = 1
+
+        ax = event.inaxes
+        # get the bounding box
+        transD2C = ax.transData.transform
+        xmin, xmax = ax.get_xlim()
+        ymin, ymax = ax.get_ylim()
+        xmin, ymin = transD2C((xmin, ymin))
+        xmax, ymax = transD2C((xmax, ymax))
+
+        # set the range
+        xp = event.x  # get event x location
+        yp = event.y  # get event y location
+        xwidth = xmax - xmin
+        ywidth = ymax - ymin
+        xcen = (xmax + xmin)*.5
+        ycen = (ymax + ymin)*.5
+        # selection center needs to be offset from zoom center
+        # to keep point under cursor stationary
+        xzc = (xp*(scl - 1) + xcen)/scl
+        yzc = (yp*(scl - 1) + ycen)/scl
+
+        bbox = [xzc - xwidth/2./scl, yzc - ywidth/2./scl, xzc +
+                xwidth/2./scl, yzc + ywidth/2./scl]
+
+        ax._set_view_from_bbox(bbox, direction=direction)
```

Comment:
Maybe I'm asking for too much, but... what about detecting a continuous wheel scroll, and saving the final view in the `ToolViewsPositions` (history) to be able to use it with the `back` and `forward` tools